### PR TITLE
https://osi-cwds.atlassian.net/browse/INT-577 DocTool Rule - R - 0247…

### PR DESCRIPTION
**Pull request by mariamcws**
…3 Default Referral Assignment and https://osi-cwds.atlassian.net/browse/INT-726  Screening HOI - Tie back-end sections together (Cases, Referrals, Screenings)

### CODE REVIEW

**Comment by longtravel** [comment]
Do we have separate endpoints for HOI for cases, referrals, and screenings, respectively?

**Comment by mariamcws** [comment]
We have separate endpoints /hoi_cases and /hoi_referrals, these return the history of involvement cases and referrals by client id. We are working on /hoi_screenings and on the endpoint /screenings/{id}/histoty_of_involvements that will return the history of involvement cases, referrals and screenings for all the clients associated with a screening

File: src/main/java/gov/ca/cwds/rest/business/rules/R02473DefaultReferralAssignment.java [hunk 1]
```diff
@@ -70,36 +71,39 @@ public R02473DefaultReferralAssignment(ScreeningToReferral screeningToReferral,
...
+      String countyCode = "00";
+
+      caseLoads = assignmentDao.findCaseLoads(screeningToReferral.getAssigneeStaffId());
+      if (caseLoads != null && caseLoads.length > 0) {
+        caseLoad = caseLoads[0];
+      }
+      if (caseLoad == null) {
+        String message = "Caseload is either inactive or on hold";
+        ServiceException se = new ServiceException(message);
+        messageBuilder.addMessageAndLog(message, se, LOGGER);
```
Comment (saleemis-cwds) [inline src/main/java/gov/ca/cwds/rest/business/rules/R02473DefaultReferralAssignment.java]: Shall we error out if there is no case load? Exception is being created but not thrown.

File: src/main/java/gov/ca/cwds/rest/business/rules/R02473DefaultReferralAssignment.java [hunk 2]
```diff
@@ -70,36 +71,39 @@ public R02473DefaultReferralAssignment(ScreeningToReferral screeningToReferral,
...
+      } else {
+        caseLoadId = caseLoad.getId();
+        countyCode = caseLoad.getCountySpecificCode();
+      }
+
+      gov.ca.cwds.rest.api.domain.cms.Assignment defaultAssignment =
+          createDefaultAssignmentToCaseLoad(countyCode, referralId,
+              screeningToReferral.getStartedAt(), caseLoadId, messageBuilder);
+      messageBuilder.addDomainValidationError(validator.validate(defaultAssignment));
```
Comment (saleemis-cwds) [inline src/main/java/gov/ca/cwds/rest/business/rules/R02473DefaultReferralAssignment.java]: Shall we error out if there are domain validation errors?

**Pull request merged**